test/LibraryManager.IntegrationTest/FileSaveRestoreTests.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+using Microsoft.Test.Apex.VisualStudio;
+using Microsoft.Test.Apex.VisualStudio.Solution;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Web.LibraryManager.IntegrationTest
+{
+    [TestClass]
+    public class FileSaveRestoreTests : VisualStudioLibmanHostTest
+    {
+        const string _projectName = @"TestProjectCore20";
+        const string _libman = "libman.json";
+
+        [TestMethod]
+        public void FileSaveRestore_AddDeleteLibrary()
+        {
+            ProjectTestExtension webProject = Solution[_projectName];
+            ProjectItemTestExtension libManConfig = webProject[_libman];
+            string projectPath = Path.Combine(SolutionRootPath, _projectName);
+
+            libManConfig.Delete();
+            LibmanTestsUtility.WaitForDeletedFile(projectPath, Path.Combine(projectPath, _libman), caseInsensitive: false, timeout: 1000);
+
+            VisualStudio.ObjectModel.Commanding.ExecuteCommand("Project.ManageClientSideLibraries");
+            LibmanTestsUtility.WaitForRestoredFile(projectPath, Path.Combine(projectPath, _libman), caseInsensitive: false, timeout: 1000);
+
+            libManConfig = webProject[_libman];
+            libManConfig.Open();
+
+            string pathToLibrary = Path.Combine(SolutionRootPath, _projectName, "wwwroot", "lib", "jquery-validate");
+            string[] expectedFiles = new[]
+            {
+                Path.Combine(pathToLibrary, "jquery.validate.js"),
+                Path.Combine(pathToLibrary, "localization", "messages_ar.js"),
+            };
+
+            Editor.Caret.MoveToExpression("\"libraries\"");
```

Comment:
This whole block of typing - now you depend on the content generated above from being constant.  Instead just replace the text buffer contents with the document you want:

```
string libmanContents = @"{
everything that goes in the file
}";

Editor.Selection.SelectAll();
Editor.KeyboardCommands.Backspace();
Editor.Edit.InsertTextInBuffer(libmanContents);
libmanConfig.Save();
```

Reply:
You should probably do this 3 times in this test:

0. Save the original contents of the file for use in (3)
1. Paste the file contents you want for the first library and verify.
2. Paste the file contents you want for the changed file (added/removed libraries) and verify.
3. Restore the original file contents from 0.

Reply:
And those 4 lines could be refactored into a method like `ReplaceFileContents(string)`

Reply:
Will do

---
In reply to: [195512080](https://github.com/aspnet/LibraryManager/pull/113#discussion_r195512080) [](ancestors = 195512080)

Reply:
Will add the function

---
In reply to: [195511264](https://github.com/aspnet/LibraryManager/pull/113#discussion_r195511264) [](ancestors = 195511264)

Reply:
will add

---
In reply to: [195512249](https://github.com/aspnet/LibraryManager/pull/113#discussion_r195512249) [](ancestors = 195512249)